tempodb/backend/s3/s3.go

```diff
@@ -421,42 +422,47 @@ func (rw *readerWriter) readRange(ctx context.Context, objName string, offset in
 	}
 }
 
-func createCore(cfg *Config, hedge bool) (*minio.Core, error) {
+func fetchCreds(cfg *Config) (*credentials.Credentials, error) {
 	wrapCredentialsProvider := func(p credentials.Provider) credentials.Provider {
 		if cfg.SignatureV2 {
 			return &overrideSignatureVersion{useV2: cfg.SignatureV2, upstream: p}
 		}
 		return p
 	}
 
-	var chain []credentials.Provider
+	chain := []credentials.Provider{
+		wrapCredentialsProvider(&credentials.Static{
+			Value: credentials.Value{
+				AccessKeyID:     cfg.AccessKey,
+				SecretAccessKey: cfg.SecretKey.String(),
```

Comment:
@joe-elliott  I was able to provide the session token via code. now this isn't longer possible.

Reply:
https://github.com/grafana/tempo/pull/3144